bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/discovery/EpsonProjectorDiscoveryParticipant.java

```diff
@@ -0,0 +1,87 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.epsonprojector.internal.discovery;
+
+import static org.openhab.binding.epsonprojector.internal.EpsonProjectorBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.sddp.SddpDevice;
+import org.openhab.core.config.discovery.sddp.SddpDiscoveryParticipant;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * Discovery Service for Epson Projectors that support SDDP.
+ *
+ * @author Michael Lobstein - Initial contribution
+ *
+ */
+@NonNullByDefault
+@Component(immediate = true)
+public class EpsonProjectorDiscoveryParticipant implements SddpDiscoveryParticipant {
+    private final Logger logger = LoggerFactory.getLogger(EpsonProjectorDiscoveryParticipant.class);
+
+    private static final String EPSON = "EPSON";
+    private static final String TYPE_PROJECTOR = "PROJECTOR";
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(SddpDevice device) {
+        final ThingUID uid = getThingUID(device);
+        if (uid != null) {
+            final Map<String, Object> properties = new HashMap<>(3);
+            final String label = device.manufacturer + " " + device.model;
+
+            properties.put(Thing.PROPERTY_MAC_ADDRESS, uid.getId());
+            properties.put(THING_PROPERTY_HOST, device.ipAddress);
+            properties.put(THING_PROPERTY_PORT, DEFAULT_PORT);
+
+            final DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                    .withRepresentationProperty(Thing.PROPERTY_MAC_ADDRESS).withLabel(label).build();
+
+            logger.debug("Created a DiscoveryResult for device '{}' with UID '{}'", label, uid.getId());
+            return result;
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(SddpDevice device) {
+        if (device.manufacturer.toUpperCase(Locale.ENGLISH).contains(EPSON)
+                && device.type.toUpperCase(Locale.ENGLISH).contains(TYPE_PROJECTOR) && !device.macAddress.isBlank()) {
+
+            logger.debug("Epson projector with mac {} found at {}", device.macAddress, device.ipAddress);
+            return new ThingUID(THING_TYPE_PROJECTOR_TCP,
+                    device.macAddress.replaceAll("-", "").toUpperCase(Locale.ENGLISH));
```

Comment:
Just to be sure, you build an UID from the MAC address all in uppercase and without ":" and "-" and this UID will then be used as MAC address in, the representation property. You apply these transformations just to be in sync with the AMX discovery ?

Note that the risk is that your discovery object in inbox or even the thing created from inbox will be regurlarly be updated switching from config parameters coming sometimes from the AMX discovery and sometimes from the SDDP discovery.

Reply:
The properties _should_ be the same. The uid/representation property is the uppercase mac address string for both (but I am unable to test because my projector does not support SSDDP). The label is different... In AMX it is "Epson Projector - **mac**" and in SDDP it is: "**device.manufacturer** -  **device.model**"